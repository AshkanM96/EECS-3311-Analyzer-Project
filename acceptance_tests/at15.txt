divide
integer_constant(1)
multiply
integer_constant(5)
integer_constant(0)
type_check
evaluate
