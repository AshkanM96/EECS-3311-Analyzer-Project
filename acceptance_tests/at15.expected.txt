  Expression currently specified: ?
  Report: Expression is initialized.
->divide
  Expression currently specified: (? / nil)
  Report: OK.
->integer_constant(1)
  Expression currently specified: (1 / ?)
  Report: OK.
->multiply
  Expression currently specified: (1 / (? * nil))
  Report: OK.
->integer_constant(5)
  Expression currently specified: (1 / (5 * ?))
  Report: OK.
->integer_constant(0)
  Expression currently specified: (1 / (5 * 0))
  Report: OK.
->type_check
  Expression currently specified: (1 / (5 * 0))
  Report: (1 / (5 * 0)) is type-correct.
->evaluate
  Expression currently specified: (1 / (5 * 0))
  Report: Error (Divisor is zero).
